odbc/src/api/get_type_info_tests.rs

```diff
@@ -0,0 +1,182 @@
+                    .unwrap()
+                    .next(None)
+            };
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetTypeInfo(handle as *mut _, data_type)
+            );
+            let stmt = (*handle).as_statement().unwrap();
+            // for each expectation, check that calling next succeeds and we get the expected value
+            expectations.iter().for_each(|name| {
+                let result = get_next(stmt);
+                assert!(result.unwrap());
+                assert_eq!(
+                    Bson::String((*name).to_string()),
+                    stmt.mongo_statement
+                        .write()
+                        .unwrap()
+                        .as_mut()
+                        .unwrap()
+                        .get_value(1)
+                        .unwrap()
+                        .unwrap()
+                )
+            });
+
+            // check there are no additional results in the set that were not expected
+            let result = get_next(stmt);
+            assert!(!result.unwrap());
+        }
+    }
+
+    #[test]
+    fn test_all_types() {
+        let expectations = vec![
+            "long",
+            "binData",
+            "array",
+            "bson",
+            "dbPointer",
+            "decimal",
+            "javascript",
+            "javascriptWithScope",
+            "maxKey",
+            "minKey",
+            "null",
+            "object",
+            "objectId",
+            "symbol",
+            "timestamp",
+            "undefined",
+            "int",
+            "bool",
+            "double",
+            "string",
+            "date",
+        ];
+        validate_result_set(SqlDataType(0), expectations);
+    }
+
+    #[test]
+    fn test_specific_type_one_response() {
+        let expectations = vec!["int"];
+        validate_result_set(SqlDataType(4), expectations);
+    }
+
+    #[test]
+    fn test_invalid_type() {
+        // note: with an empty list of expectations, this will just check that calling next fails
+        validate_result_set(SqlDataType(95), vec![])
+    }
+
+    #[test]
+    fn test_invalid_cursor_state_error() {
+        // checks for invalid cursor state when calling get_value before next
+        let handle: *mut _ = &mut MongoHandle::Statement(Statement::with_state(
+            std::ptr::null_mut(),
+            StatementState::Allocated,
+        ));
+        unsafe {
+            let stmt = (*handle).as_statement().unwrap();
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetTypeInfo(handle as *mut _, SqlDataType(4))
+            );
+            let value = stmt
+                .mongo_statement
+                .write()
+                .unwrap()
+                .as_ref()
+                .unwrap()
+                .get_value(1);
+            assert!(value.is_err());
+        }
+    }
+
+    #[test]
+    fn test_get_all_values() {
```

Comment:
I think that I would rename it to `test_get_all_type_info_row_values` just to disambiguate what "values" means here.